app/lib/actions/foreman_ansible/play_hostgroup_roles.rb

```diff
@@ -0,0 +1,65 @@
+module Actions
+  module ForemanAnsible
+    # Actions that initiaztes the playbook run for roles assigned to
+    # the host. It doest that either locally or via a proxy when available.
+    class PlayHostgroupRoles < Actions::EntryAction
+      include ::Actions::Helpers::WithContinuousOutput
+      include ::Actions::Helpers::WithDelegatedAction
+
+      def plan(hostgroup, proxy_selector = ::ForemanAnsible::ProxySelector.new)
+        if hostgroup.hosts.empty?
+          raise ::Foreman::Exception.new(N_('hostgroup is empty'))
+        end
+        input[:hostgroup] = { :id => hostgroup.id, :name => hostgroup.name }
+        proxy = proxy_selector.determine_proxy(hostgroup.hosts[0])
```

Comment:
Understood - we should provide a way to choose strategies too, no need to do that in this PR. Let's keep this discussion for foreman-dev or further PRs. Keep it as is now for 1.4.0 😄